sockeye/arguments.py

```diff
@@ -15,14 +15,58 @@
 Defines commandline arguments for the main CLIs with reasonable defaults.
 """
 import argparse
+import json
 import os
 import sys
+import types
 from typing import Callable, Optional
 
 from sockeye.lr_scheduler import LearningRateSchedulerFixedStep
 from . import constants as C
 from . import data_io
 
+# Adapted from https://stackoverflow.com/questions/28579661/getting-required-option-from-namespace-in-python
+class ConfigArgumentParser(argparse.ArgumentParser):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.argument_definitions = {}
+        self.argument_actions = []
+        self._overwrite_add_argument(self)
+        self.add_argument("--config", help="Config file in JSON format.", type=str)  # Not FileType so that we can get the path here
+
+    def register_argument(self, _action, *args, **kwargs):
```

Comment:
will arguments also be registered if there is no group?

Reply:
Yes they will. Line 34 calls _overwrite_add_argument on itself for supporting the non-group case. What is not supported (but not used in sockeye) are sub-parsers [and possibly other exotic argparse use cases].